File: src/main/java/ru/lewis/libraryloader/model/Dependency.java
package ru.lewis.libraryloader.model;

public class Dependency {

    private final String group;
    private final String artifact;
    private final String version;

    public Dependency(String group, String artifact, String version) {
        this.group = group;
        this.artifact = artifact;
        this.version = version;
    }

    public String getGroup()    { return group; }
    public String getArtifact() { return artifact; }
    public String getVersion()  { return version; }

    public String toPath() {
        return group.replace('.', '/') + "/" + artifact + "/" + version + "/" + artifact + "-" + version + ".jar";
    }

    public String toFileName() { return artifact + "-" + version + ".jar"; }

    public String toPomPath() {
        return group.replace('.', '/') + "/" + artifact + "/" + version + "/" + artifact + "-" + version + ".pom";
    }

    @Override public String toString() { return group + ":" + artifact + ":" + version; }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Dependency d)) return false;
        return toString().equals(d.toString());
    }

    @Override public int hashCode() { return toString().hashCode(); }

    public static Dependency of(String notation) {
        String[] parts = notation.split(":");
        if (parts.length != 3) throw new IllegalArgumentException("Invalid dependency: " + notation);
        return new Dependency(parts[0], parts[1], parts[2]);
    }
}

File: src/main/kotlin/ru/lewis/libraryloader/model/Dependency.kt
package ru.lewis.libraryloader.model

data class Dependency(
    val group: String,
    val artifact: String,
    val version: String
) {
    fun toPath(): String {
        val groupPath = group.replace('.', '/')
        return "$groupPath/$artifact/$version/$artifact-$version.jar"
    }

    fun toFileName(): String = "$artifact-$version.jar"

    fun toPomPath(): String {
        val groupPath = group.replace('.', '/')
        return "$groupPath/$artifact/$version/$artifact-$version.pom"
    }

    override fun toString(): String = "$group:$artifact:$version"

    companion object {
        fun of(notation: String): Dependency {
            val parts = notation.split(":")
            require(parts.size == 3) { "Invalid dependency: $notation" }
            return Dependency(parts[0], parts[1], parts[2])
        }
    }
}